rasterio/_shim1.pyx

```diff
@@ -25,6 +27,19 @@ cdef GDALDatasetH open_dataset(
 
     # Note well: driver choice, open options, and sibling files
     # are not supported by GDAL versions < 2.0.
+
+    if allowed_drivers:
+        raise GDALOptionNotImplementedError(
+            "Driver selection is not implemented in GDAL 1.x")
```

Comment:
Like this @geowurster ?

Reply:
Perfect 🙌 !

Reply:
Thanks for pushing me on this: it's the right thing and my initial inclination to just quietly ignore (or warn) was wrong. Someone should right a blog post about this kind of issue in software development :)

Reply:
Someone with an active blog + readers perhaps?